example/progress.js

```diff
@@ -2,16 +2,25 @@ const path = require('path');
 const fs   = require('fs');
 const ytdl = require('..');
 
-var url = 'https://www.youtube.com/watch?v=WhXefyLs-uw';
-var output = path.resolve(__dirname, 'video.mp4');
+const url = 'https://www.youtube.com/watch?v=WhXefyLs-uw';
+const output = path.resolve(__dirname, 'video.mp4');
 
-var video = ytdl(url);
+const video = ytdl(url);
+let starttime;
 video.pipe(fs.createWriteStream(output));
-video.on('progress', function(chunkLength, downloaded, total) {
+video.once('response', () => {
+  starttime = Date.now();
+})
+video.on('progress', (chunkLength, downloaded, total) => {
+  const float_downloaded = downloaded / total;
+  const downloaded_min = (Date.now() - starttime) / 1000 / 60;
   process.stdout.cursorTo(0);
   process.stdout.clearLine(1);
-  process.stdout.write((downloaded / total * 100).toFixed(2) + '% ');
+  process.stdout.write(`${(float_downloaded * 100).toFixed(2)}% downloaded`);
+  process.stdout.write(`(${downloaded / 1024 / 1025}MB of ${total / 1024 / 1024}MB)`);
+  process.stdout.write(`, running for: ${downloaded_min.toFixed(2)}minutes`);
+  process.stdout.write(`, estimated time left: ${(downloaded_min / float_downloaded - downloaded_min).toFixed(2)}minutes`);
```

Comment:
Since this line is really long, it won't be printed in one line some terminals if there isn't enough room. It will instead print everything in several new lines.

I did some research on how to continuously print to multiple lines, here's something that worked with the [readline module](https://nodejs.org/api/readline.html):

```js
  readline.cursorTo(process.stdout, 0);
  process.stdout.write(`${(float_downloaded * 100).toFixed(2)}% downloaded `);
  process.stdout.write(`(${downloaded / 1024 / 1025}MB of ${total / 1024 / 1024}MB)\n`);
  process.stdout.write(`running for: ${downloaded_min.toFixed(2)}minutes`);
  process.stdout.write(`, estimated time left: ${(downloaded_min / float_downloaded - downloaded_min).toFixed(2)}minutes`);
  readline.moveCursor(process.stdout, 0, -1);
```